### PR TITLE
Add webpack-dev-server

### PR DESCRIPTION
**Pull request by bttf**
It works

## It working
Screenshot 2023-03-09 at 3 51 15 PM

## Tested production build
Screenshot 2023-03-09 at 3 52 12 PM

### CODE REVIEW

File: src/devtools/controller.ts [hunk 1]
```diff
@@ -5,6 +5,8 @@ import {
   SetOverridesMessage,
 } from "../../devtools";
 
+const isProd = process.env.NODE_ENV === "production";
```
Comment (tinahollygb) [inline src/devtools/controller.ts:8]: What if we had something like:

```ts
const isChrome = typeof chrome !== 'undefined'
```

My concern with removing the dev build is that production builds take longer to build than dev, so I wouldn't want us to remove the ability to create a dev build.

I'd rather we retain the ability to build both and instead add a script like `serve:devtools` where we serve the devtools. Similarly for popup and options, we'd need a `serve:options` and `serve:popup` or maybe we can have a single webpack-dev-server run serve all 3. Or maybe we serve them all on the same server at the same host but at different paths, e.g. `/devtools.html`, `popup.html`, `options.html`.

File: src/devtools/controller.ts [hunk 2]
```diff
@@ -57,3 +62,26 @@ export function setOverrides(data: Omit<SetOverridesMessage, "type">) {
...
+        defaultValue: false,
+        rules: [],
+      },
+    },
+    experiments: {},
+    attributes: {
+      test: 123,
+    },
+    overrides: {},
+  };
```
Comment (tinahollygb) [inline src/devtools/controller.ts:80]: I wouldn't use mock data in the real app. This is why something like Storybook comes in handy. When a component is structured using presentational-container component architecture, and the features are passed into the main component, e.g. `<DevTools />` (which I think is just `<App />` right now, and probably needs to change), then we can mock it as needed in the prototyping tools.

File: src/devtools/controller.ts (hunk 1 above)
Comment (tinahollygb) [reply src/devtools/controller.ts:8]: Regarding this, I see in the screenshot that the dev server hosts all 3, disregard that part.

**Review by tinahollygb — CHANGES_REQUESTED** [review]
I think we need the webpack-dev-server to work but I don't agree with the approach taken:

- I don't think we should disable dev builds as production takes too long. Can you restore that and instead check for `chrome`? 

Minor concerns:

- I don't think we should add hardcoded mock data into the source code. Do you think it's possible to re-architect it to not have such a dependency on `chrome` and allow us to pass in all dependencies via props? 

See inline comments for more info.

**Comment by bttf** [comment]
@tinahollygb Just an update, definitely have ways to address your concerns. Thanks for the feedback. Just waiting for the dust to settle on visual editor updates and will tackle these as soon as that's done.

**Comment by bttf** [comment]
This has gone stale. Deleting

**Pull request closed without merging**